### PR TITLE
DEV: Delay rendering sidebar sections after sidebar is shown

### PR DESCRIPTION
**Pull request by nattsw**
Rendering sidebar sections is costly due to 1. the popper and 2. the number of sections that may be displayed.

This PR introduces a `DeferredRender` which allows us to split the time to paint both.

Before:

https://github.com/discourse/discourse/assets/1555215/1eee16c9-8439-426a-8dec-47d6ac229962

After:

https://github.com/discourse/discourse/assets/1555215/c872aee9-9875-4de5-9812-5d8759baa32c

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/deferred-render.gjs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  constructor() {
+    super(...arguments);
+    runAfterFramePaint(() => (this.shouldRender = true));
+  }
+
+  <template>
+    {{#if this.shouldRender}}
+      {{yield}}
+    {{else}}
+      {{loadingSpinner}}
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/deferred-render.gjs:18]: so it tries to display a loading spinner for the duration of one frame paint? wouldn't it be better to display nothing at all?

Comment (nattsw) [reply app/assets/javascripts/discourse/app/components/deferred-render.gjs]: Yeah fair 👍

File: app/assets/javascripts/discourse/app/components/sidebar/hamburger-dropdown.gjs [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/components/sidebar/hamburger-dropdown.gjs:2]:
```suggestion
```

Comment (nattsw) [reply app/assets/javascripts/discourse/app/components/sidebar/hamburger-dropdown.gjs:2]: Screenshot 2024-05-29 at 10 17 35 AM


 😿 my auto linter is so lame..

**Pull request merged**